x/staging/subspaces/types/subspace.go

```diff
@@ -92,9 +92,9 @@ func (sub Subspace) Validate() error {
 // SubspaceTypeFromString convert a string in the corresponding SubspaceType
 func SubspaceTypeFromString(subType string) (SubspaceType, error) {
```

Comment:
Is it really necessary? Where is this used?

Reply:
This will be used inside cli commands to convert the strings passed into the correspondent subspace type

Reply:
There's no need to implement this by hand using constant values. We can simply rely on Protobuf generated maps as Cosmos does [here](https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/types/vote.go#L89)

Reply:
Didn't see that! thanks

Reply:
@RiccardoM I found out that they use also [this function](https://github.com/cosmos/cosmos-sdk/blob/66ee994ce4b5951a373131f0ee5bb0436f507ab4/x/gov/client/utils/utils.go#L10) in conjunction with the one you pointed inside CLI. I don't get the point in using 2 functions instead one. Just getting the value from the first one is not feasible since users should write `SUBSPACE_TYPE_OPEN` inside the tx

Reply:
@bragaz They're normalizing the values so that the users do not have to care about casing when writing the option. With that function, `Yes` and `yes` are both valid, while if there was no such function then only `Yes` would be considered valid. 

I think we can implement that function as well to make it easier for users to use the CLI.

Also, we can remove the `SUBSPACE_TYPE_UNSPECIFIED` option as the subspace will always have a known type (either `open` or `closed`).

Reply:
The problem of casing can be solved by using `strings.ToLower()` on the value the the user gives so it will always be something like `open` or `close` when checking.
Also, the `SUBSPACE_TYPE_UNSPECIFIED` is useful when someone tries to edit the subspace and don't want to edit that field, so the default value will be that one and it will tells that the field shouldn't be edited.

Reply:
@bragaz I think it's better to follow the Cosmos way here. Otherwise we would have to always remember to call the `string.ToLower` function on the given value. In this case we can simply use the `SubspaceTypeFromString` and do the normalization inside there. Maybe even using `strings.ToLower` inside it would be a good choice. 

Also, I think that instead of using a default value of `SUBSPACE_TYPE_UNSPECIFIED`, we should stick with `[do-not-edit]` for the subspace type as well. This would make it easier for a user to understand since it would have the same behavior of other flags